dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,475 @@
+        """
+        This function requires an active dask.distribtued client as it uses
+        the dask.distributed's concurrent.futures API.
+
+        This function implicitly assumes that higher scores are better. Note:
+        this matters if you are using a scoring function that measures loss,
+        where lower values imply a better model.
+
+        This function works with classes with a certain API, not only
+        scikit-learn estimators. It is assumed that the class
+
+        * implements ``partial_fit(self, X, y, **kwargs)``.
+        * implements ``set_params(self, **kwargs) -> self``
+        * implements ``score(self, X, y) -> number``
+        * supports ``sklearn.base.clone(self, safe=True)``. Support for this
+          function comes down to having an function ``get_params(self)``.
+
+        """
+        self.scorer_, _ = self._check_scorer(self.model)
+
+        client = _get_client()
+        if client:
+            import distributed
+            variables = {'val_score': -np.inf, 'config': None}
+
+            shared = {}
+            for key, value in variables.items():
+                # distributed.Variable requires client
+                shared[key] = distributed.Variable(key)
+                shared[key].set(value)
+        else:
+            shared = None
+            warn('Tried to set a distributed variable to allow '
+                  'partial results to be preserved. Having a '
+                  'distributed client will resolve this (though this '
+                  'will print validation loss and the config)')
+
+        X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.1)
+        data = {'train': [X_train, y_train], 'val': [X_val, y_val]}
+
+        if client:
+            for key in data.keys():
+                data[key][0] = client.scatter(data[key][0])
+                data[key][1] = client.scatter(data[key][1])
+        else:
+            warn('Tried to scatter data with the distributed '
+                 'client, which may help relieve memory pressure. '
+                 'Having a distributed client will resolve this')
+
+        # now start hyperband
+        R, eta = self.R, self.eta
+        s_max = math.floor(math.log(self.R, self.eta))
+        B = (s_max + 1) * self.R
+        all_kwargs = []
+        for s in reversed(range(s_max + 1)):
+            n = math.ceil((B / R) * eta**s / (s + 1))
+            r = R * eta ** -s
+            all_kwargs += [{'s': s, 'n': n, 'r': r, 'dry_run': dry_run,
+                            'eta': eta, '_prefix': f's={s}', 'shared': shared,
+                            'scorer': self.scorer_}]
+
+        if self.n_jobs == -1:
+            futures = [dask.delayed(_successive_halving)(self.params, self.model,
+                                                         data, **kwargs, **fit_kwargs)
+                       for kwargs in all_kwargs]
+            results = [f.compute() for f in futures]
```

Comment:
Here also you should avoid calling compute many times in succession and instead use the following:

    results = dask.compute(futures)

Also, we might want to avoid using the term futures here, as that has a special meaning in Dask.  Perhaps `delayed_results` or something similar?